go/service/chat_local.go

```diff
@@ -328,6 +329,9 @@ func (h *chatLocalHandler) GetThreadNonblock(ctx context.Context, arg chat1.GetT
 			SessionID: arg.SessionID,
 			Thread:    remoteThread,
 		})
+
+		// This means we transmitted with success, so cancel local thread
+		cancel()
```

Comment:
I could also just stick this at the end
```
select {
case <-bctx.Done():
default:
    cancel()
}
```

but I figured that would just get handled once the context died.